lib/web/users.go

```diff
@@ -104,7 +104,16 @@ func createUser(r *http.Request, m userAPIGetter, createdBy string) (*ui.User, e
 
 	user.SetRoles(req.Roles)
 
-	updateUserTraits(req, user)
+	// checkAndSetDefaults makes sure either TraitsPreset
+	// or AllTraits field to be populated. Since empty
+	// AllTraits is also used to delete all user traits,
+	// we explicitly check if TraitsPreset is empty so
+	// to prevent traits deletion.
+	if req.TraitsPreset == (traitsPreset{}) {
```

Comment:
That's a strange check, can you make it a pointer and then check against nil? And/or have an isEmpty method on the struct that returns true/false?

Reply:
:D
Made the field pointer so now it can be compared with `nil` https://github.com/gravitational/teleport/pull/42620/commits/1a39fd7dee18702b11d3049dc6f72d303496be99